mountpoint-s3/src/fs/config.rs

```diff
@@ -130,14 +133,33 @@ impl CacheConfig {
                 serve_lookup_from_cache: true,
                 file_ttl: TimeToLive::INDEFINITE_DURATION,
                 dir_ttl: TimeToLive::INDEFINITE_DURATION,
+                negative_cache_ttl: TimeToLive::INDEFINITE_DURATION,
                 ..Default::default()
             },
             TimeToLive::Duration(ttl) => Self {
                 serve_lookup_from_cache: true,
                 file_ttl: ttl,
                 dir_ttl: ttl,
+                negative_cache_ttl: ttl,
                 ..Default::default()
             },
         }
     }
+
+    pub fn with_negative_cache_ttl(self, negative_cache_ttl: TimeToLive) -> Self {
+        match negative_cache_ttl {
+            TimeToLive::Minimal => Self {
+                negative_cache_ttl: Self::default().negative_cache_ttl,
```

Comment:
Just setting the TTL here may not be enough. We use the same `serve_lookup_from_cache` flag both for the "positive" cache and the negative cache ([here](https://github.com/awslabs/mountpoint-s3/blob/b8338af6019cbe2cbecdccf2805edafefef6a766/mountpoint-s3/src/superblock.rs#L792), in particular). I suspect we need to introduce a separate flag and ensure it is set to false on `Minimal` here.

Reply:
I still see an issue here where even when setting `--negative-metadata-ttl minimal`, the negative cache is still needlessly populated with a very short TTL. My suggestion is to add an independent flag to `CacheConfig`, e.g. `use_negative_cache`, and use it in `update_from_remote` to replace `serve_lookup_from_cache` (`mountpoint-s3/mountpoint-s3/src/superblock.rs:792`).